### ISSUE TITLE
HTML reports are unclear; can't tell which assertions are triggering a build failure

### ISSUE THREAD

**Issue by Lange** [GoogleChrome/lighthouse-ci#232]
Howdy,

I recently set up LHCI for the first time on a project. I have set a number of the assertions to be either `off` or `warn`. However, the HTML report still shows them as normal, which no indication that I have reduced their severity in my `lighthouserc.js` file.

What this means is that when I click the "Details" link on my PR to view the HTML report from LHCI, I see a bunch of errors and warnings but I cannot tell which ones I have ignored in my `lighthouserc.js` file, and therefore I can't tell which ones I actually need to fix to get CI to pass.

Adjusting the UI of these HTML reports to make it clearer what actually needs to be changed in order for the build to not be marked as a failure would be great.

Thanks,
Alex

**Comment by patrickhulce** [comment]
Thanks for filing @Lange! The HTML report is just the regular lighthouse report that was collected and isn't specific to Lighthouse CI. It sounds like you would be interested in Lighthouse CI mutating the report to remove anything you didn't assert before it gets uploaded.

A few more questions then...

- Are you still interested in this if assertion details aren't displayed anywhere in the report? (This is the Lighthouse core report that has no concept of assertions)
- Are you still interested in this if the error state in the report doesn't match the assertion? (The scores and pass/warn/fail states in the report are going to be based on Lighthouse's values)
- How do you want the audits with assertions that are off to be handled in the report, be not applicable? passing? not there at all?

Perhaps I'm wrong, but I think at the end of the day what you're asking for is a completely separate UI for the Lighthouse CI assertion results that focuses on what you asserted and how it didn't meet that bar. That's what the Lighthouse CI server UI is for and on our roadmap, but it's unlikely to ever become a standalone report like the path you're using now.

**Comment by Lange** [comment]
Interesting, I was under the impression that the server was just a way for me to get around the 7-day limitation of the free `temporary-public-storage` upload target. I was not aware that it actually provided a different UI and was more geared towards reporting assertions.

Is just spinning up an instance of this server what I want to do instead? Does having this server change what the "Details" button on the per-PR status reports links to?

**Comment by Lange** [comment]
Oh sorry, and to answer your questions:

> Are you still interested in this if assertion details aren't displayed anywhere in the report? (This is the Lighthouse core report that has no concept of assertions)

I am unsure what you mean by this. Can you elaborate?

> Are you still interested in this if the error state in the report doesn't match the assertion? (The scores and pass/warn/fail states in the report are going to be based on Lighthouse's values)

Possibly, but I'd want some kind of indicator or explanation about the two different error states. Like if the report shows an `error` for HTTP2, but my LHCI config has `use-http2` set to `off`, I'd want the report to be like "hey this is a thing that you _could_ improve, but it is an ignored assertion right now and isn't the reason your PR is marked as having a failing build".

> How do you want the audits with assertions that are off to be handled in the report, be not applicable? passing? not there at all?

The information about those assertions is still valuable to see, so I'd prefer if they are still present in the report. I think "not applicable" might be the most suitable status. One solid example of this is the HTTP2 assertion -- I serve my static site via Netlify which handles all of that for me, so it kind of doesn't matter if my tests use HTTP2 or not. It isn't something that my codebase is responsible for, and therefore is not a unit under test.

**Comment by patrickhulce** [comment]
> Is just spinning up an instance of this server what I want to do instead?

Yes, *eventually*. Right now the UI isn't exactly what you want. But it's the *only* Lighthouse CI specific UI that exists so that's where the features you want will eventually appear :)


> Does having this server change what the "Details" button on the per-PR status reports links to?

Yes, uploading to a server means your GH "Details" links will now be to the Lighthouse CI UI instead of the plain Lighthouse report.

> Can you elaborate?

The current "Details" link you're looking at is just a regular Lighthouse report that has no knowledge of Lighthouse CI, your assertions, etc. The only thing we can really do from within Lighthouse CI to modify this UI is fudge the JSON data to get a different-looking regular Lighthouse report. New error states, suggestions about what's ignored, etc really won't be possible and would only come to the Lighthouse CI server UI.
